src/utils.cpp

```diff
@@ -1102,25 +1101,40 @@ bool symbol_has_cpp_mangled_signature(const std::string &sym_name)
     return false;
 }
 
-pid_t parse_pid(const std::string &str)
+static std::string get_invalid_pid_message(const std::string &pid,
+                                           const std::string &msg)
 {
+  return "pid '" + pid + "' " + msg;
+}
+
+std::optional<pid_t> parse_pid(const std::string &str, std::string &err)
```

Comment:
Would be good to add unit tests for this but it's out of scope of this PR

Reply:
We have 4 runtime tests for this but yeah, unit tests would be good too.